tests/unit/test__main.py

```diff
@@ -207,3 +327,194 @@ def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
     )
     assert mock_run.call_count == 4
     assert e_info.value.args[0] == "Some of the validations raised an exception"
+
+
+@mock.patch(
+    "argparse.ArgumentParser.parse_args",
+    return_value=argparse.Namespace(**VALIDATE_COLUMN_CONFIG),
+)
+@mock.patch(
+    "data_validation.__main__.build_config_managers_from_args",
+    return_value=[
+        config_manager.ConfigManager(
+            VALIDATE_COLUMN_CONFIG, MockIbisClient(), MockIbisClient(), verbose=False
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_successful_column_validation_with_mocked_run_validation(
+    mock_args, mock_build, mock_run
+):
+    main.main()
+
+
+@mock.patch(
+    "argparse.ArgumentParser.parse_args",
+    return_value=argparse.Namespace(**BROKEN_VALIDATE_COLUMN_CONFIG_MISSING_COMMAND),
+)
+@mock.patch(
+    "data_validation.__main__.build_config_managers_from_args",
+    return_value=[
+        config_manager.ConfigManager(
+            BROKEN_VALIDATE_COLUMN_CONFIG_MISSING_COMMAND,
+            MockIbisClient(),
+            MockIbisClient(),
+            verbose=False,
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_throws_for_malformed_input_config_missing_command(
+    mock_args, mock_build, mock_run
+):
+    with pytest.raises(Exception) as e_info:
+        main.main()
+    assert e_info.value.args[0] == "'Namespace' object has no attribute 'command'"
+
+
+@mock.patch(
+    "argparse.ArgumentParser.parse_args",
+    return_value=argparse.Namespace(**BROKEN_VALIDATE_COLUMN_CONFIG_INCORRECT_COMMAND),
+)
+@mock.patch(
+    "data_validation.__main__.build_config_managers_from_args",
+    return_value=[
+        config_manager.ConfigManager(
+            BROKEN_VALIDATE_COLUMN_CONFIG_INCORRECT_COMMAND,
+            MockIbisClient(),
+            MockIbisClient(),
+            verbose=False,
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_throws_for_malformed_input_config_incorrect_command(
+    mock_args, mock_build, mock_run
```

Comment:
Arg order.
I won't add any more of these because it's probably annoying. I'm sure you can find the rest yourself :-)